File: app/src/main/java/com/absinthe/anywhere_/adapter/manager/WrapContentStaggeredGridLayoutManager.java
package com.absinthe.anywhere_.adapter.manager;

import androidx.recyclerview.widget.RecyclerView;
import androidx.recyclerview.widget.StaggeredGridLayoutManager;

import timber.log.Timber;

public class WrapContentStaggeredGridLayoutManager extends StaggeredGridLayoutManager {
    public WrapContentStaggeredGridLayoutManager(int spanCount, int orientation) {
        super(spanCount, orientation);
    }

    @Override
    public void onLayoutChildren(RecyclerView.Recycler recycler, RecyclerView.State state) {
        try {
            super.onLayoutChildren(recycler, state);
        } catch (IndexOutOfBoundsException e) {
            Timber.e("encounter an IOOBE in RecyclerView");
        }
    }
}

File: app/src/main/java/com/absinthe/anywhere_/adapter/manager/WrapContentStaggeredGridLayoutManager.kt
package com.absinthe.anywhere_.adapter.manager

import androidx.recyclerview.widget.RecyclerView
import androidx.recyclerview.widget.RecyclerView.Recycler
import androidx.recyclerview.widget.StaggeredGridLayoutManager
import timber.log.Timber

class WrapContentStaggeredGridLayoutManager(spanCount: Int, orientation: Int) : StaggeredGridLayoutManager(spanCount, orientation) {

    override fun onLayoutChildren(recycler: Recycler, state: RecyclerView.State) {
        try {
            super.onLayoutChildren(recycler, state)
        } catch (e: IndexOutOfBoundsException) {
            Timber.e("encounter an IOOBE in RecyclerView")
        }
    }
}